openstack/identity/v3/roles/urls.go

```diff
@@ -2,14 +2,24 @@ package roles
 
 import "github.com/gophercloud/gophercloud"
 
+const (
+	rolePath    = "roles"
+	userPath    = "users"
+	projectPath = "projects"
+)
+
 func listURL(client *gophercloud.ServiceClient) string {
-	return client.ServiceURL("roles")
+	return client.ServiceURL(rolePath)
 }
 
 func getURL(client *gophercloud.ServiceClient, roleID string) string {
-	return client.ServiceURL("roles", roleID)
+	return client.ServiceURL(rolePath, roleID)
 }
 
 func listAssignmentsURL(client *gophercloud.ServiceClient) string {
 	return client.ServiceURL("role_assignments")
```

Comment:
nit: consider moving `role_assignments` to const as well.

Reply:
I'm not sure any of the `urls.go` files have constants. I might be wrong, but it's worth double-checking. If there isn't precedence, I'd suggest not using constants here.

Reply:
In other `urls.go`, I found that any paths used more than once is defined as const and it's not otherwise.

Reply:
> In some other urls.go, I found that any paths used more than once is defined as const and it's not otherwise.

```shell
$ find . -name urls.go -exec grep -l const {} \;
```

Well I guess there's a good amount of `const`'s being used. In various ways, too: some have all parts of the URL as a constant, some have only repeated parts mixed with strings.

I don't want to spend too much time nit-picking this one so it's totally up to you. 😄

Reply:
Cool. I'll leave it as is atm 😄

Reply:
Sounds good. Let me know when this PR is good to go for a full review. I saw a few messages going back and forth and may have prematurely jumped in :)

Reply:
I think that it's good to go for a full review :)